packages/@ember/-internals/glimmer/tests/integration/helpers/custom-helper-test.js

```diff
@@ -658,6 +658,40 @@ moduleFor(
 
       this.assertText('huzza!');
     }
+
+    ['@test class-based helper gives helpful warning when mutating a value that was tracked already']() {
+      this.add(
+        'helper:hello-world',
+        class extends Helper {
+          compute() {
+            this.get('value');
+            this.set('value', 123);
+          }
+        }
+      );
+
+      expectWarning(() => {
```

Comment:
IMHO, this should be a deprecation (not a warning). We ultimately want to move this to an error condition (whereas with warnings, they would live on forever without a path to remove).

Reply:
I wasn’t sure about `until`, which is why I made it a warn

Reply:
Set the `until` to `4.0.0`